clippy_lints/src/loops/mut_range_bound.rs

```diff
@@ -135,21 +134,17 @@ impl BreakAfterExprVisitor {
 }
 
 impl<'tcx> Visitor<'tcx> for BreakAfterExprVisitor {
-    fn visit_expr(&mut self, expr: &'tcx Expr<'tcx>) {
-        if self.past_candidate {
-            return;
-        }
-
+    type Result = ControlFlow<()>;
+    fn visit_expr(&mut self, expr: &'tcx Expr<'tcx>) -> ControlFlow<()> {
         if expr.hir_id == self.hir_id {
             self.past_expr = true;
         } else if self.past_expr {
             if matches!(&expr.kind, ExprKind::Break(..)) {
                 self.break_after_expr = true;
             }
 
-            self.past_candidate = true;
-        } else {
-            intravisit::walk_expr(self, expr);
+            return ControlFlow::Break(());
         }
+        intravisit::walk_expr(self, expr)
```

Comment:
This was previously inside of an `else {}` but isn't anymore with this change, so we descend into things when we don't need to right now